### ISSUE TITLE
[Ubuntu 17.04] gpd-update hanging with 100% cpu for 27 hours now.

### ISSUE THREAD

**Issue by evilsocket** [cawilliamson/ansible-gpdpocket#49]
I know the kernel should take a while to build, but compilation is over and 27 hours are really **a lot** even for the GPD.

![gpd_0](https://user-images.githubusercontent.com/86922/29244497-69497998-7fb9-11e7-96d3-b127815f484c.png)

![gpd_2](https://user-images.githubusercontent.com/86922/29244501-80e1e3ec-7fb9-11e7-9c09-d31d02927292.png)

![file_000](https://user-images.githubusercontent.com/86922/29244507-cdef3aea-7fb9-11e7-8ca4-6d23e990a0a8.jpeg)

I'll keep it running in the meantime, just in case, but it is very very weird :P

PS: **THANK YOU** for this project

**Comment by cawilliamson** [comment]
Thanks for the images - very useful!

Yeah - that kind of timeframe is crazy! The build time of a kernel on the pocket is around 2-3hrs.

I'd abort that and try again - possibly after a reboot. Looks like an apt issue - intel-microcode is *tiny* and shouldn't take longer than a few seconds to install.

**Comment by evilsocket** [comment]
Yeah I built a kernel in the past and that's the time usually on such hardware :D

Ok, rebooting and trying again ... will update asap.

**Comment by cawilliamson** [comment]
That having been said - this looks more like an apt/ubuntu quirk than an issue with the playbooks so I'm going to close this one off with WONTFIX for now but will leave it open pending your update. 👍

**Comment by evilsocket** [comment]
Mmmm well, this is happening because the script is installing and updating custom packages which have little to do with Ubuntu itself, I'm using the very same distro and configuration, with the same type of packages on other machines and this is the first time I see a problem like that ...

**Comment by cawilliamson** [comment]
Custom packages? Little to do with Ubuntu... what are you talking about?

Every single package being installed is from the Ubuntu repos dude - there's no "custom packages" and if you're running an Intel CPU on any rig without intel-microcode installed - you really shouldn't be.

Since I don't see this discussion going anywhere productive and it looks like you're having problems with apt and not with my ansible playbooks - I'm going to close this one and wish you the best of luck with your GPD Pocket.

**Comment by evilsocket** [comment]
lol ok dude, whatever, but calm down ok? I was just reporting you a problem. I hope it's just me and it doesn't happen to other people.

**Comment by evilsocket** [comment]
I didn't read the whole repo code, giving the compilation of a kernel I thought there were custom kernel and packages, that's all.

Bye.

**Comment by evilsocket** [comment]
why reopened? wasn't this just a problem I had which had nothing to do with your ansible playbooks?

**Comment by cawilliamson** [comment]
@evilsocket Wanted to take a minute to apologise - the response above wasn't called for and certainly wasn't in the spirit of open source nor the GPD Pocket community. Sorry about that!

I just saw your post on reddit and wanted to re-open this one to see if we can resolve the issue working together.

I'm completely mystified by this one to be honest and nobody else has reported an issue with this particular package. What happens if you try to install the package manually?

`apt-get -y install intel-microcode`

It's likely an easy fix (or workaround if necessary) but it would sure be good to resolve it.

**Comment by cawilliamson** [comment]
So you'll use my work, pay me nothing, insult me and basically tell me to fuck off and expect your post not to get deleted?
